### PR TITLE
TLS support

### PR DESCRIPTION
**Pull request by hannesm**

### CODE REVIEW

**Comment by hannesm** [comment]
inspiration came from mirage-www (which afaics is a bit overcomplicated) amongst other repositories... (the interesting bit is that conduit does not provide anything to just listen on port x, and continue with code to listen on port y and z)... thus doing this manually for now

**Comment by hannesm** [comment]
(and this is live at http://hannes.nqsb.io / https://hannes.nqsb.io - push also works :))

File: canopy_dispatch.ml [hunk 1]
```diff
@@ -0,0 +1,108 @@
+open Lwt.Infix
+
+type store_ops = {
+  subkeys : string list -> string list list Lwt.t ;
+  value : string list -> string option Lwt.t ;
+  update : unit -> string list Lwt.t
+}
```
Comment (hannesm) [inline canopy_dispatch.ml:7]: this is the piece I'm not happy abou... but I failed to properly construct te `module Store` and pass it to `Canopy_dispatch`...

Comment (abbysmal) [inline canopy_dispatch.ml:7]: That'll do for know, I have an idea on how to this but that's fine enough like this for the moment. :)

**Comment by hannesm** [comment]
what's holding you off from merging?

**Comment by abbysmal** [comment]
I was not too sure if you were going to add documentation (as discussed this weekend) but that can come later I guess. :)
I'll merge right away if you want, everything looks good to me. :)

**Comment by hannesm** [comment]
pls merge, I'll provide docs later today :)

**Comment by abbysmal** [comment]
Ok!
Thank you for your hard work, merged. :)

**Pull request merged**